include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php

```diff
@@ -0,0 +1,195 @@
+?>
+                    <tr>
+                        <td scope="row" >
+                            <input name="smbd_tax_terms[]" value="<?php echo $term->slug; ?>" type="checkbox" class="terms">
+                        </td>
+                        <td>
+                            <label for="smbd_tax_terms"><?php echo $term->name; ?> (<?php echo $term->count . ' '; _e( 'Posts', 'bulk-delete' ); ?>)</label>
+                        </td>
+                    </tr>
+<?php
+				}
+?>
+            </table>
+<?php
+			}
+?>
+            <table class="optiontable">
+				<?php
+				$this->render_filtering_table_header();
+				$this->render_restrict_settings();
+				$this->render_delete_settings();
+				$this->render_limit_settings();
+				$this->render_cron_settings();
+				?>
+            </table>
+
+            </fieldset>
+<?php
+			$this->render_submit_button();
+		} else {
+?>
+            <h4><?php _e( "This WordPress installation doesn't have any non-empty custom taxonomies defined", 'bulk-delete' ) ?></h4>
+<?php
+		}
+	}
+
+	protected function convert_user_input_to_options( $request, $options ) {
+		$options                       = array();
+		$options['post_type']          = bd_array_get( $request, 'smbd_tax_post_type', 'post' );
+		$options['selected_taxs']      = bd_array_get( $request, 'smbd_taxs' );
+		$options['selected_tax_terms'] = bd_array_get( $request, 'smbd_tax_terms' );
+		$options['restrict']           = bd_array_get_bool( $request, 'smbd_taxs_restrict', false );
+		$options['private']            = bd_array_get_bool( $request, 'smbd_taxs_private' );
+		$options['limit_to']           = absint( bd_array_get( $request, 'smbd_taxs_limit_to', 0 ) );
+		$options['force_delete']       = bd_array_get_bool( $request, 'smbd_taxs_force_delete', false );
+
+		$options['date_op']            = bd_array_get( $request, 'smbd_taxs_op' );
+		$options['days']               = absint( bd_array_get( $request, 'smbd_taxs_days' ) );
+
+		return $options;
+	}
+
+	public function delete( $delete_options ) {
+		if ( bd_array_get( $_POST, 'smbd_taxs_cron', 'false' ) == 'true' ) {
+			$freq = $_POST['smbd_taxs_cron_freq'];
+			$time = strtotime( $_POST['smbd_taxs_cron_start'] ) - ( get_option( 'gmt_offset' ) * 60 * 60 );
+
+			if ( $freq == -1 ) {
+				wp_schedule_single_event( $time, Bulk_Delete::CRON_HOOK_TAXONOMY, array( $delete_options ) );
+			} else {
+				wp_schedule_event( $time, $freq, Bulk_Delete::CRON_HOOK_TAXONOMY, array( $delete_options ) );
+			}
+			$msg = __( 'Posts from the selected custom taxonomies are scheduled for deletion.', 'bulk-delete' ) . ' ' .
+				sprintf( __( 'See the full list of <a href = "%s">scheduled tasks</a>' , 'bulk-delete' ), get_bloginfo( 'wpurl' ) . '/wp-admin/admin.php?page=' . Bulk_Delete::CRON_PAGE_SLUG );
+		} else {
+			return $deleted_count = $this->delete_posts_by_taxonomy( $delete_options );
+		}
+	}
+
+	public static function delete_posts_by_taxonomy( $delete_options ) {
```

Comment:
This should not be either public or static.

Also, we can replace this method with just the build_query method.

Look into either DeletePostsByCategoryMetabox or DeletePostsByTagMetabox to see how to do it. This is something that I recently added.